packages/devtools_flutter/lib/src/connect_screen.dart

```diff
@@ -92,15 +97,33 @@ class _ConnectScreenBodyState extends State<ConnectScreenBody> {
         ),
         RaisedButton(
           child: const Text('Connect'),
-          onPressed: connect,
+          onPressed: _connect,
         ),
       ],
     );
   }
 
-  void connect() {
-    // TODO(https://github.com/flutter/devtools/issues/1112): Connect to a VM.
-    Navigator.pushNamed(context, '/connected');
+  Future<void> _connect([_]) async {
+    var connected = false;
+    try {
+      connected = await service.FrameworkCore.initVmService(
+        '',
+        explicitUri: Uri.parse(controller.text),
+      );
+    } catch (_) {
+      Scaffold.of(context).showSnackBar(
+        SnackBar(
+          content: Text(
+            'Unable to connect to Dart VM at "${controller.text}". '
+            'Please specify a running Dart VM URL.',
+          ),
+        ),
+      );
+    }
+    if (connected) {
+      final uriQuery = 'uri=${Uri.encodeQueryComponent(controller.text)}';
+      return Navigator.popAndPushNamed(context, '/?$uriQuery');
+    }
   }
```

Comment:
Q: is it possible for connected to be false but doesn't throw any exception at all that results in the SnackBar not showing up?

Reply:
I believe it is!

Take a look at the implementation of framework_core's initVmService: https://github.com/flutter/devtools/blob/012aec0eb672b8289d6e9bf1d750174405f71a6f/packages/devtools_app/lib/src/framework/framework_core.dart#L35-L63

I've changed this to always pop the snackbar when the method returns false.

Excellent catch!

Reply:
Yay, awesome!